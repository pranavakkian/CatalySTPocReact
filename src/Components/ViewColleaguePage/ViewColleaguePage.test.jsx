import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ViewColleague from './index';

describe('ViewColleaguePage', () => {
  const mockStore = configureMockStore();
  test('Rendered ViewColleaguePageSuccessfully', () => {
    const store = mockStore(() => ({
      ColleagueReducer: {
        data: [
          {
            name: 'Abc',
            jobTitle: 'Engineer',
            contact: '12345',
          },
        ],
      },
    }));
    const { container } = render(<Provider store={store}><ViewColleague /></Provider>);
    expect(container).toMatchSnapshot();
  });
  test('Checking for no data in store', () => {
    const store = mockStore(() => ({
      ColleagueReducer: {
        data: [],
      },
    }));
    render(<Provider store={store}><ViewColleague /></Provider>);
    expect(screen.getByText('Data Not Available')).toBeInTheDocument();
  });
});
