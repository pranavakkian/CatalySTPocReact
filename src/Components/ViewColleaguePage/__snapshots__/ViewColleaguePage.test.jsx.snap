// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewColleaguePage Rendered View Colleague Page Successfully 1`] = `
<div>
  <div>
    <h1
      class="ui center aligned header"
    >
      Colleague Information
      <div
        class="ui divider"
      />
      <table
        class="ui compact table"
      >
        <thead
          class=""
        >
          <tr
            class=""
          >
            <th
              class=""
            >
              Employee
            </th>
            <th
              class=""
            >
              Job Title
            </th>
            <th
              class=""
            >
              Contact
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class=""
            >
              --
            </td>
            <td
              class=""
            >
              --
            </td>
            <td
              class=""
            >
              --
            </td>
          </tr>
        </tbody>
      </table>
    </h1>
  </div>
</div>
`;
